### PR TITLE
Added code coverage pipeline

### PR DESCRIPTION
**Pull request by rs--**

### CODE REVIEW

File: .jenkins/scripts/global/init-access.sh
Comment (CyanDevs) [inline .jenkins/scripts/global/init-access.sh:5]: I've been thinking if this should load the OE library just to use `helpers.WaitForAptLock()` as this only covers one potential lock.

File: .jenkins/scripts/global/init-docker.sh
Comment (CyanDevs) [inline .jenkins/scripts/global/init-docker.sh:12]: Please do not do this as it is a big security vulnerability

File: .jenkins/scripts/global/init-docker.sh
Comment (CyanDevs) [inline .jenkins/scripts/global/init-docker.sh:1]: In my opinion, while this is fine, it is an awkward bash script to have. It does not fully encapsulate the docker setup (e.g. missing docker install) and complicates a series of simple commands with the need for passing args and parsing params.

File: .jenkins/scripts/global/init-install.sh
Comment (CyanDevs) [inline .jenkins/scripts/global/init-install.sh:1]: Could improve readability by having a line per package.

File: .jenkins/scripts/global/make-world.sh
Comment (CyanDevs) [inline .jenkins/scripts/global/make-world.sh:1]: Use `$(command)` instead for POSIX compatibility

File: .jenkins/scripts/solutions/azure-config.sh
Comment (CyanDevs) [inline .jenkins/scripts/solutions/azure-config.sh:4]: Use `$(command)` for POSIX compatibility

File: .jenkins/scripts/solutions/azure-config.sh
Comment (CyanDevs) [inline .jenkins/scripts/solutions/azure-config.sh:1]: If you separate this script from the az-cli repo setup, then perhaps this line would be better in the bash script that initializes the apt repo. Then in this script you either assume azure-cli is installed, or do a check to make sure it is installed before proceeding.

File: .jenkins/scripts/solutions/init-config.sh
Comment (CyanDevs) [inline .jenkins/scripts/solutions/init-config.sh:1]: Regarding this file name `solutions/init-config.sh`: this file isn't initialization for solutions test per se. It is for setting up the apt repo for azure-cli, which is a dependency only due to the fact that we're using an alternative authentication method in Jenkins for some solution tests.

File: .jenkins/scripts/global/init-docker.sh
Comment (rs--) [reply .jenkins/scripts/global/init-docker.sh:12]: It's a copy and paste from an existing pipeline, not sure if it's needed for the test to work

File: .jenkins/scripts/solutions/init-config.sh
Comment (rs--) [reply .jenkins/scripts/solutions/init-config.sh:1]: good point, I'll clean up the naming

File: .jenkins/scripts/global/init-access.sh
Comment (rs--) [reply .jenkins/scripts/global/init-access.sh:5]: The functionality of this file was already split into different scripts, I forgot to update this one.

File: .jenkins/scheduled-tests.Jenkinsfile
Comment (rs--) [reply .jenkins/scheduled-tests.Jenkinsfile:19]: Given that dotnet is no longer an option, I think this will actually be a lot cleaner as there are fewer differences between the nightly and code coverage pipelines

File: .jenkins/scripts/global/make-tests.sh
Comment (CyanDevs) [inline .jenkins/scripts/global/make-tests.sh:1]: Curious -- why echo continue on failure?

File: .jenkins/scripts/global/wait-dpkg.sh
Comment (CyanDevs) [inline .jenkins/scripts/global/wait-dpkg.sh:1]: We should change this to 
 our more thorough script, but since we're out of time for this migration we should leave it to later. https://github.com/openenclave/openenclave/blob/master/.jenkins/library/vars/helpers.groovy#L13 or https://github.com/openenclave/openenclave/blob/master/scripts/ansible/install-ansible.sh#L10

**Comment by CyanDevs** [comment]
As this moves the existing Jenkinsfiles we will need to make sure we change the job configurations to use the new location of the Jenkinsfiles when we merge this in.

File: .jenkins/scripts/global/make-tests.sh
Comment (rs--) [reply .jenkins/scripts/global/make-tests.sh:1]: This is a temporary change; the tests were being flakey and I wanted to bypass the failed tests

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (CyanDevs) [inline .jenkins/pipelines/scheduled-tests.Jenkinsfile:106]: Are both SQL solution for USEAST and CANADACENTRAL necessary? Nightly tests and CC tests only ran one of CANADACENTRAL or USEAST (respectively). Running both will add an hour to the test.

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (CyanDevs) [inline .jenkins/pipelines/scheduled-tests.Jenkinsfile:179]: cat on a workspace will return exit code 1

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (CyanDevs) [inline .jenkins/pipelines/scheduled-tests.Jenkinsfile:177]: Can publish to a storage blob. An example is: https://github.com/deislabs/mystikos/blob/main/.jenkins/release.Jenkinsfile#L199-L205

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (CyanDevs) [inline .jenkins/pipelines/scheduled-tests.Jenkinsfile:26]: This would be set as well when the nightly pipeline is ran, yet the original nightly pipeline does not set this. What are the implications? If there are any concerns it would be better to set this only when running the Code Coverage pipeline.

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (CyanDevs) [inline .jenkins/pipelines/scheduled-tests.Jenkinsfile:3]: Nit: Since your intention is to replace the nightly pipeline too, please also mention the original nightly pipeline here

**Review by CyanDevs — COMMENTED** [review]
Looks good now, just a couple remaining comments to address before merging.

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (rs--) [reply .jenkins/pipelines/scheduled-tests.Jenkinsfile:26]: This file has the possibility of replacing the nightly pipeline; I think we should attempt that outside of this PR. I'll investigate the differences when/if we end up replacing the nightly pipeline with this file.

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (rs--) [reply .jenkins/pipelines/scheduled-tests.Jenkinsfile:105]: I'll use USEAST for this pipeline

**Review by CyanDevs — APPROVED** [review]
LGTM. Thanks Rob!

File: .jenkins/pipelines/scheduled-tests.Jenkinsfile
Comment (salsal97) [inline .jenkins/pipelines/scheduled-tests.Jenkinsfile:11]: we might only need a 300 min timeout - afraid if a 10 hr timeout would cause extra resource usage in case something is stuck?

**Pull request merged**
